### ISSUE TITLE
sci-misc/boinc - no menu buttons in boincmgr with x11-libs/wxGTK-2.8.12.0

### ISSUE THREAD

**Issue by romw** [BOINC/boinc-ng#1152]
**Reported by toralf on 29 Jun 42134688 22:33 UTC**
The origin is here : https://bugs.gentoo.org/show_bug.cgi?id=379979 .
It worked flawlessly with x11-libs/wxGTK-2.8.11.0

The version is 6.12.42 - but current subversion trunk show same behavior.

**Comment by romw** [comment]
**Commented by toralf on 23 Jan 42145948 18:11 UTC**
It is not a bug in wxGTK : http://trac.wxwidgets.org/ticket/14032#comment:7

**Comment by romw** [comment]
**Commented by Trog Dog on 28 Apr 42183682 21:25 UTC**
workaround is to resize the BM window, if this helps - workaround shamelessly stolen from redhat bugzilla

**Comment by romw** [comment]
**Commented by toralf on 16 Oct 42183713 09:53 UTC**
yes - confirmed, that this works.

**Comment by romw** [comment]
**Commented by Trog Dog on 10 May 42255058 01:56 UTC**
Replying to [toralf](comment:3):
> yes - confirmed, that this works.
Recompiling against 2.8.11.0 works (will post on gentoo bugzilla change to make to ebuild). For the BOINC devs - something changes between wxwidgets 2.8.11 and 2.8.12 that brings about these missing menus.

**Comment by romw** [comment]
**Commented by Trog Dog on 10 Aug 42381887 02:06 UTC**
From my inbox from user Huib
```
Hi Mister Dog,
 
Adding a SendSizeEvent(); after installing the menubar (CAdvancedFrame::CreateMenu) seems to fix the problem with the disappearing menubar.
 
Groetjes,
 Huibert.
```

**Comment by romw** [comment]
**Commented by Trog Dog on 15 Jun 42384522 21:17 UTC**

```
Based on release 2.0.27
 

Index: clientgui/AdvancedFrame.cpp
 ===================================================================
 --- clientgui/AdvancedFrame.cpp (revision 25677)
 +++ clientgui/AdvancedFrame.cpp (working copy)
 @@ -701,6 +701,10 @@

 wxMenuBar* m_pOldMenubar = GetMenuBar();
 SetMenuBar(m_pMenubar);
 +#ifdef __WXGTK__
 + /* Force WX to recalculate the size of the (new) menu bar */
 + SendSizeEvent();
 +#endif
 #ifdef __WXMAC__
 m_pMenubar->MacInstallMenuBar();
 MacLocalizeBOINCMenu();

 
Groetjes,
 Huibert
```

Patch supplied by Huibert (Huib on the boinc forums)

**Comment by romw** [comment]
**Commented by Trog Dog on 5 Feb 42385296 01:04 UTC**
Actually still current with 7.0.28

**Comment by romw** [comment]
**Commented by Trog Dog on 7 Jul 42385300 15:28 UTC**
Applied patch and tested on 5 gentoo boxes (compiled 7.0.28) and it addresses the problem.

**Comment by romw** [comment]
**Commented by romw on 14 Apr 42389252 10:08 UTC**
(In [25702])     - MGR: Fix menu redraw issue under Ubuntu's new interface.
      fixes #1180
      (From Huibert)

    clientgui/
        AdvancedFrame.cpp

**Comment by romw** [comment]
**Commented by toralf on 13 Apr 42389602 14:12 UTC**
The comment in the fix indicates it is Ubuntu specific - but it is an issue under Gentoo too.

**Comment by romw** [comment]
**Commented by toralf on 1 Apr 42466378 17:40 UTC**
seems not fixed under an almost stable Gentoo - I still have to apply the patch against 7.0.29

**Comment by romw** [comment]
**Commented by romw on 19 Apr 42490956 03:25 UTC**
(In [25812])     - MGR: Fix menu redraw issue under Ubuntu's new interface.
      fixes #1180
      (From Huibert)

    clientgui/
        AdvancedFrame.cpp

**Comment by romw** [comment]
**Commented by toralf on 7 Jan 42491199 20:25 UTC**
The issue is still in 7.0.29 here at as table Gentoo Linux - therefore I apply always this patch :
tfoerste@n22 /usr/local/portage/sci-misc $ cat boinc/files/wxwidgets-2.8.0.12-menufix.patch 
--- clientgui/AdvancedFrame.cpp_orig    2012-05-20 00:01:49.441702516 +0200
+++ clientgui/AdvancedFrame.cpp 2012-05-20 00:03:00.136490955 +0200
@@ -701,6 +701,10 @@
 
     wxMenuBar* m_pOldMenubar = GetMenuBar();
     SetMenuBar(m_pMenubar);
+#ifdef __WXGTK__
+/* Force WX to recalculate the size of the (new) menu bar */
+    SendSizeEvent();
+#endif
 #ifdef __WXMAC__
     m_pMenubar->MacInstallMenuBar();
     MacLocalizeBOINCMenu();
